blog/storage.md

```diff
@@ -0,0 +1,390 @@
+title: SuperMQ Storage
+
+# Introduction
+
+Building a modern IoT messaging platform is no small feat. And when that platform is engineered to be distributed, highly scalable, and secure, the complexity increases exponentially. Why? The goal quickly shifts from just handling a high volume of messages to managing an ecosystem that connects countless users and millions of devices interacting through a mix of communication protocols. So, how do you manage your most valuable asset - your data - at this scale?
+
+A common temptation is to throw a single, massive database at the problem. But that’s a recipe for disaster -  a classic “jack of all trades, master of none” that quickly buckles under pressure. It not only creates performance bottlenecks, but also limits scalability, and opens the door to security risks. The truth is, the data flowing through a modern IoT messaging platform isn’t all the same. It comes in many forms - structured, unstructured, relational, temporal - and each type has its own specific needs.
+
+That is where SuperMQ takes a smarter path. Rather than forcing every kind of data into a single model, it embraces Polyglot Persistence - a strategy that is as practical as it is powerful. The idea is simple: use the right tool for the job. So, how does SuperMQ put this into practice? It breaks down its data needs like this:
+
+- Core Metadata: It includes entities like users, clients (devices or applications), groups, and channels.
+- Event Data: It consists of the billions of messages published by clients, where each message is an immutable event stamped with a precise time.
+- Authorization Data: It defines who can perform what action on which resource and requires a model that can efficiently represent and query relationships with very low latency.
+
+The table below provides a high-level overview of the above strategy, which will be explored in detail in the subsequent sections.
+
+| Data Domain              | Selected Technology |
+|--------------------------|---------------------|
+| Core Entity Metadata     | PostgreSQL          |
+| Time-Series Message Data | TimescaleDB         |
```

Comment:
Should we use other dbs we are using something https://github.com/absmach/supermq-contrib/tree/main/cmd